models.py

```diff
@@ -43,6 +55,15 @@ def generate_auth_token(self, expiration = 60000):
         s = Serializer(app.config['SECRET_KEY'], expires_in = expiration)
         return s.dumps({ 'user_id': self.user_id })
 
+    def make_provider(self):
+        self.is_provider = True
+
+    def udpate_rating(self, rating):
+        rat = self.rating * self.num_ratings
```

Comment:
Don't we need to guard rating with some range like [0-5] inclusive? I am okay with this logic of updating ratings for now, but we might have to refine this in future.